### DISCUSSION TITLE
[QA] CLI improvement

### DISCUSSION THREAD

**Discussion by vanessayuenn** [storybookjs/storybook#25692] [Announcement]
#### Feature owner: @JReinhold 
Please comment in this discussion thread for any bug report and feedback for this feature as you test out Storybook 8. Kindly adhere to the following format:

> **Project**: Name of your project as entered in Notion
> **Version**: The version of storybook you tested with
> **Bug/feedback**: Details of the bugs/feedback you have

Feature owner should triage the comments here frequently and open issues as needed.

**Comment by JReinhold** [comment]
**Project**: SvelteUI
**Version**: 8.0.0-alpha.13
**Bug/feedback**:

The upgrade command changed the format in `package.json`, going from tab indentation to 2-spaced indentation throughout the file.

**Reply by vanessayuenn** [reply]
https://github.com/storybookjs/storybook/issues/25812

**Comment by JReinhold** [comment]
**Project**: SvelteUI
**Version**: 8.0.0-alpha.13
**Bug/feedback**:

The upgrade command didn't warn/error about not being able to find the main configuration file. The configuration is in a non-standard directory which the user must define with the `--config-dir` flag, but there was no prompt for this. This caused all automigrations to be skipped.

**Reply by vanessayuenn** [reply]
https://github.com/storybookjs/storybook/issues/25814

**Comment by valentinpalkovic** [comment]
**Project:**: https://github.com/storybookjs/addon-onboarding
**Version**: 8.0.0-alpha.14
**Bug/feedback:**: False alarm `UpgradeStorybookToSameVersionError`

Running `pnpm dlx storybook@next upgrade` in `@storybook/addon-onboarding` throws with the following error, although the project uses Storybook dependencies with the version `7.1.0-alpha.38`. Everything works fine when using `npx storybook@next upgrade`:

```
SB_CLI_UPGRADE_0004 (UpgradeStorybookToSameVersionError): You are trying to upgrade Storybook to the same version that is currently installed in the project, version 8.0.0-alpha.14. This is not supported.

This usually happens when running the upgrade command without a version specifier, e.g. "npx storybook upgrade".
This will cause npm to run the globally cached storybook binary, which might be the same version that you already have.
This also happens if you're running the Storybook CLI that is locally installed in your project.

If you intended to upgrade to the latest version, you should always run the Storybook CLI with a version specifier to force npm to download the latest version:

"npx storybook@latest upgrade"

If you intended to re-run automigrations, you should run the "automigrate" command directly instead:

"npx storybook@8.0.0-alpha.14 automigrate"
```

**Reply by JReinhold** [reply]
Filed here https://github.com/storybookjs/storybook/issues/25734

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

`storybook@next upgrade` does not upgrade satellite packages, leading to duplicated dependencies (with possibly different major versions) after upgrading.

Details and repro are discussed in the issue:
https://github.com/storybookjs/storybook/issues/25731

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

There's a bug in `storybook add` where if you specify the version of the addon, it doubles its version and fails.

**Repro:**
1. Go to any Storybook project
2. If you run `npx storybook@next add @storybook/addon-themes` it will fail because it will (I don't know why) try to install version 8.0.0 of that addon, and there's currently no v8.0.0 (makes sense)
<img width="539" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/298cdb5e-ada4-4a69-9cf2-62b26c2e6a7c">

3. If you then run `npx storybook@next add @storybook/addon-themes@8.0.0-alpha.14`, it will try to install `@storybook/addon-themes@8.0.0-alpha.14@^8.0.0-alpha.14`. (first, it probably shouldn't use the caret, second there's the parsing issue)
<img width="539" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/cf309bc8-d506-4e11-b7a7-9868e45e6798">

**Reply by vanessayuenn** [reply]
tracked here https://github.com/storybookjs/storybook/issues/25811

**Comment by valentinpalkovic** [comment]
**Project**: React Vite sandbox with yarn pnp
**Version**: 8.0.0-alpha.15
**Bug/feedback**: Caret is not added to the package.json

After installation, exact version numbers are added to the `package.json` instead of versions with a `caret` specifier.

**Reply by vanessayuenn** [reply]
tracked here: https://github.com/storybookjs/storybook/issues/25811

**Comment by valentinpalkovic** [comment]
**Project**: React Vite sandbox with yarn pnp
**Version**: 8.0.0-alpha.14
**Bug/feedback**: Upgrading Storybook from version 0.0.0 to version 8.0.0-alpha.14

It seems that the initial Storybook version cannot be detected. I guess it has something to do with yarn pnp mode.

![image](https://github.com/storybookjs/storybook/assets/5889929/9e6dbf9c-1b8c-4ca9-a9b7-23c956d8d6f7)

**Reply by JReinhold** [reply]
This should be fixed now, could you try again?

**Comment by valentinpalkovic** [comment]
**Project**: React Vite sandbox with yarn pnp
**Version**: 8.0.0-alpha.14
**Bug/feedback**: specific upgrade leads to versions with a caret

Running `yarn dlx storybook@8.0.0-alpha.14 upgrade` puts Storybook dependencies into the package.json **with** a caret, although it should be listed without.

**Reply by vanessayuenn** [reply]
tracked here: https://github.com/storybookjs/storybook/issues/25811

**Comment by valentinpalkovic** [comment]
**Project**: React Vite sandbox with yarn pnp
**Version**: 8.0.0-alpha.14
**Bug/feedback**: UpgradeStorybookToSameVersionError not shown

Running `yarn dlx storybook@8.0.0-alpha.14 upgrade` on the project, which has the latest prerelease version installed, will not through the `UpgradeStorybookToSameVersionError`. I think this is connected to the above issue, where the version cannot be detected.

**Comment by valentinpalkovic** [comment]
Related issue: https://github.com/storybookjs/storybook/issues/25770

**Comment by jonniebigodes** [comment]
**Project**: Any (it affects core frameworks)
**Version: 8.0.0-alpha.14**
**Bug/feedback:** Adding satellite addons via the `add` command fails to install


If a user runs the `add` command to enable the Webpack-based packages mentioned in the following [documentation](https://storybook.js.org/docs/8.0/builders/webpack) (i.e., `@storybook/addon-webpack5-compiler-babel` and `@storybook/addon-webpack5-compiler-swc`) leads to the CLI failing to install them with the following errors:

![swc-compiler-installation-issue-latest](https://github.com/storybookjs/storybook/assets/22988955/1cb3c81c-dc3b-44d0-8242-b3bd626c1a15)

![swc-compiler-installation-issue-next](https://github.com/storybookjs/storybook/assets/22988955/fde1a4cd-018c-48d3-b9e9-440bbca23366)

Based on my understanding of Storybook's CLI, it will try and match the version specified and tie it to the addon's version. Since there's no `latest`  or `next` versions (e.g., 7.6, 8.0 ), it will error out.

One caveat that I checked was that suffixing the package with the latest version (i.e., `npx storybook@latest add @storybook/addon-webpack5-compiler-swc@1.0.0`) it works

![swc-compiler-suffixed-version-ok](https://github.com/storybookjs/storybook/assets/22988955/002109e9-edda-4bca-853c-2bf48141b240)

However, replacing `latest` with `next` and suffixing the package version, it errors out.

![swc-compiler-next-nok](https://github.com/storybookjs/storybook/assets/22988955/030cba16-a149-433e-8b9d-3c6e58ca7c67)


This will probably have compound effects on other packages, such as the [coverage addon](https://storybook.js.org/docs/writing-tests/test-coverage#set-up-the-coverage-addon). 

If we're committed to introducing the `add` command as the method a user should install addons, we'll need to get a better understanding of what's behind this and how best to address it. 

In the interim, I would appreciate some feedback on this so that the documentation can be adjusted accordingly, as I worked on them alongside with @valentinpalkovic. Thanks in advance


cc @shilman , @yannbf

**Reply by valentinpalkovic** [reply]
Filed here https://github.com/storybookjs/storybook/issues/25790
